include/vrv/scoringupfunctor.h

```diff
@@ -0,0 +1,100 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        scoringupfunctor.h
+// Author:      Martha E. M. Thomae Elias
+// Created:     2024
+// Copyright (c) Authors and others. All rights reserved.
+/////////////////////////////////////////////////////////////////////////////
+///
+#ifndef __VRV_SCORINGUPFUNCTOR_H__
+#define __VRV_SCORINGUPFUNCTOR_H__
+
+#include "functor.h"
+
+namespace vrv {
+
+//----------------------------------------------------------------------------
+// ScoringUpFunctor
+//----------------------------------------------------------------------------
+
+/**
+ * This class...
+ */
+class ScoringUpFunctor : public Functor {
+public:
+    /**
+     * @name Constructors, destructors
+     */
+    ///@{
+    ScoringUpFunctor();
+    virtual ~ScoringUpFunctor() = default;
+    ///@}
+
+    /*
+     * Abstract base implementation
+     */
+    bool ImplementsEndInterface() const override { return true; }
+
+    /*
+     * Methods
+     */
+    /**
+     * @name: Divide the notes of a voice into sequences to be processed individualy
+     */
+    ///@{
+    std::vector<std::vector<std::pair<LayerElement *, data_DURATION>>> SubdivideSeq(
+        std::vector<std::pair<LayerElement *, data_DURATION>> dursInVoiceSameMensur);
```

Comment:
Please double check the way you are passing the vector:
* If you are only reading it (which I  assume), pass it as `const std::vector<> &dursInVoiceSameMensur`
* If you are modifying it in place, pass is as `std::vector<> &dursInVoiceSameMensur`
This avoids for a copy to be made every time you pass it.

Check this for **all** methods

Reply:
Done! Thank you :)

Should I do the same with the `pair` parameter in the `GetNumberValue` function? Or did you wanted this only for the vectors since these are longer?

https://github.com/martha-thomae/verovio/blob/6834c9abbbf28197d0b48026206172bfab6c241c/include/vrv/scoringupfunctor.h#L59-L61

Reply:
Yes, there too. Otherwise calling the function makes a complete copy of the sequence. By passing it as a reference (with `&`) but making in `const` you can avoid a copy to be made but also ensure that it is not going to be modified in the function.